src/context/context.ts

```diff
@@ -39,8 +39,8 @@ export class Context {
     this.environments = env
     this.strictVariables = renderOptions.strictVariables ?? this.opts.strictVariables
   }
-  public getRegister (key: string) {
-    return (this.registers[key] = this.registers[key] || {})
+  public getRegister (key: string, defaultValue?: any) {
+    return (this.registers[key] = this.registers[key] || (defaultValue ?? {}))
```

Comment:
We can eliminate the use of `defaultValue ?? {}` by setting `defaultValue = {}` in parameter line. So it's actually a *final* default value.

As mentioned before, I still don't understand why we need change this method.

Reply:
Sorry. I must have been half asleep. 😞